.github/workflows/black.yml

```diff
@@ -0,0 +1,19 @@
+name: black
+on: [push]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1  # Use v1, NOT v2
+      - uses: actions/setup-python@v1
+      - run: pip install black
+      - run: black --check .
+      - name: If needed, commit black changes to a new pull request
+        if: failure()
+        run: |
+          black .
+          git config --global user.name github-actions
+          git config --global user.email '${GITHUB_ACTOR}@users.noreply.github.com'          
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
+          git commit -am "fixup! Format Python code with psf/black push"
+          git push --force origin HEAD:$GITHUB_REF
```

Comment:
Thank you for your contribution. Can you change this to use `ruff` and also use it in the pre-commit hooks?